sensor/kubernetes/listener/resources/convert.go

```diff
@@ -334,7 +352,9 @@ func (w *deploymentWrap) populateImageIDs(pods ...*v1.Pod) {
 				if features.LocalImageScanning.Enabled() {
 					// imageutil.IsInternalImage requires Sensor to already know about the OpenShift internal registries,
 					// which is ok because Sensor listens for Secrets before it starts listening for Deployment-like resources.
-					image.IsClusterLocal = imageutil.IsInternalImage(image.GetName())
+					image.IsClusterLocal = imageutil.IsInternalImage(image.GetName(), &imageutil.IsInternalImageOptions{
```

Comment:
What is the difference between `w.registryStore` and the singleton? 🤔

Reply:
the singleton is meant to be used for all non-unit-testing purposes. Otherwise, a non-singleton registry store should be used for more control over how it is populated